vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -2401,6 +2475,85 @@ public FooterRow appendFooterRow() {
         return insertColumnLayer(getLastFooterLayerIndex() + 1).asFooterRow();
     }
 
+    /**
+     * Removes the footer row from the grid.
+     *
+     * @see #removeAllFooterRows()
+     * @param footerRow
+     *            the footer row to remove
+     * @throws NoSuchElementException
+     *             if the footer row cannot be found
+     */
+    public void removeFooterRow(FooterRow footerRow) {
```

Comment:
Accidentally ran into this strange behavior: if I append two footer rows and then remove one, the column order of grid changes (used BeanGridPage.java):

```java
grid.appendFooterRow();
var row = grid.appendFooterRow();
grid.removeFooterRow(row);
```

https://github.com/vaadin/flow-components/assets/1222264/48f788dc-dc2a-42e9-83f8-a0b085fb880c

Reply:
This issue doesn't seem to relate to the new changes and since it's an edge case, we can leave it for later.